aten/src/ATen/core/ivalue.cpp

```diff
@@ -122,6 +122,19 @@ std::ostream& printDict(
 }
 }
 
+// Properly disambiguate the type of an empty list
+std::ostream& printMaybeAnnotatedDict(
+    std::ostream& out,
+    const IValue& the_dict,
+    IValueFormatter formatter) {
+  if (the_dict.toGenericDict().size() == 0) {
```

Comment:
I think this function & the list function both have a bug. A list with an optional value type needs to be disambiguated as well. `x : Dict[Int, Optional[Int] = {1: None}`. It's possible this can't be triggered yet with Lists yet because we still check the types of lists as opposed to check if all values are insertable as constants. But I believe it can with dictionaries. See: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/python_print.cpp#L799